src/drivers/intel/cavs/spi.c

```diff
@@ -426,36 +447,66 @@ struct sspi *sspi_get(uint32_t type)
 	return NULL;
 }
 
+void spi_push(void *data, size_t size)
+{
+	struct sspi *spi = sspi_get(SOF_SPI_INTEL_SLAVE);
+	struct spi_dma_config *config = spi->config + SPI_DIR_TX;
+
+	spi->ipc_status = IPC_WRITE;
+
+	do {
+		size_t chunk_size = min(size, SPI_BUFFER_SIZE);
+
+		rmemcpy(config->src_buf, data, chunk_size);
+		dcache_writeback_region(config->src_buf, chunk_size);
+		sspi_trigger(spi, SSPI_TRIGGER_START, SPI_DIR_TX);
+		data += chunk_size;
+		size -= chunk_size;
+		gpio_write(GPIO14, 1);
+		delay(1);
+		gpio_write(GPIO14, 0);
+		/* 0.5s */
```

Comment:
if those GPIO toggles are for debug only use a helper function that is commented out if it's not needed?